hooks/useWhiteList.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from "react"
+import { whitelistedAddresses } from "@/lib/whitelistAddresses"
+import useConnectedWallet from "@/hooks/useConnectedWallet"
+const useWhitelist = () => {
+  const { externalWallet } = useConnectedWallet()
+  const [isWhitelisted, setIsWhitelisted] = useState(false)
+
+  useEffect(() => {
+    checkIfWhitelisted()
+    if (externalWallet?.address) {
```

Comment:
why have this check if you're already calling `checkIfWhitelisted`  prior on line 9?